File: AccountTransferMvpApp/UnitTests/src/com/gaddzeit/utils/tests/unit/ServiceLocatorTests.java
package com.gaddzeit.utils.tests.unit;

import com.gaddzeit.presenters.IPresenter;
import com.gaddzeit.utils.PresenterTypeEnum;
import com.gaddzeit.utils.ServiceLocator;
import org.junit.Test;

import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertTrue;

public class ServiceLocatorTests {

    @Test
    public void givenSut_whenResolveMethodIsPassedAnEnumValue_thenMatchingPresenterIsReturned() {
        PresenterTypeEnum presenterTypeEnum = PresenterTypeEnum.AccountTransfer;
        IPresenter accountTransferPresenter = ServiceLocator.resolve(presenterTypeEnum);
        assertNotNull("The returned type should not be null", accountTransferPresenter);
        assertTrue("The expected return type should be returned.", accountTransferPresenter instanceof IPresenter);
    }
}

File: AccountTransferMvpApp/AccountTransferMvpAppTests/Utilts.Tests.Unit/ServiceLocatorTests.m
#import "ServiceLocatorTests.h"
#import "PresenterTypeEnum.h"
#import "ServiceLocator.h"
#import "AccountTransferPresenter.h"

@implementation ServiceLocatorTests

-(void)testGivenSut_whenResolveMethodIsPassedAnEnum_thenTheMatchingTypeIsInstantiatedAndReturned {
    PresenterTypeEnum presenterTypeEnum = AccountTransfer;
    id accountTransferPresenter = [ServiceLocator resolve:presenterTypeEnum];

    STAssertTrue([accountTransferPresenter isKindOfClass:[AccountTransferPresenter class]], @"The expected return type was not returned.");
}

@end
